android-components/components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt

```diff
@@ -8,109 +8,229 @@ import android.graphics.Rect
+): Boolean {
+    // Measure menu.
     val spec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
     containerView.measure(spec, spec)
 
-    val (availableHeightToTop, availableHeightToBottom) = getMaxAvailableHeightToTopAndBottom(anchor)
-    val (availableWidthToLeft, availableWidthToRight) = getMaxAvailableWidthToLeftAndRight(anchor)
-    val containerHeight = containerView.measuredHeight
+    val containerRecyclerView =
+        containerView.findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
+    val recyclerViewAdapter = containerRecyclerView.adapter
+
+    if (recyclerViewAdapter == null) {
+        Log.log(
+            priority = Log.Priority.ERROR,
+            tag = "mozac-popupwindow",
+            message = "The browser menu recyclerview has no adapter set.",
+        )
+        return false
+    }
 
-    val fitsUp = availableHeightToTop >= containerHeight
-    val fitsDown = availableHeightToBottom >= containerHeight
-    val reversed = availableWidthToLeft < availableWidthToRight
+    val containerViewHeight = containerView.measuredHeight
+    val recyclerViewHeight = containerRecyclerView.measuredHeight
+    val recyclerViewItemCount = recyclerViewAdapter.itemCount
+
+    val hasViewInitializedCorrectly =
+        containerViewHeight > 0 && recyclerViewHeight > 0 && recyclerViewItemCount > 0
+    if (!hasViewInitializedCorrectly) {
+        Log.log(
+            priority = Log.Priority.ERROR,
+            tag = "mozac-popupwindow",
+            message = "containerViewHeight: $containerViewHeight must be more than 0," +
+                "recyclerViewHeight: $recyclerViewHeight must be more than 0 &" +
+                "recyclerViewItemCount: $recyclerViewItemCount must be more than 0.",
+        )
+        return false
+    }
 
     // Popup window does not need a input method. This avoids keyboard flicker when menu is opened.
     inputMethodMode = PopupWindow.INPUT_METHOD_NOT_NEEDED
 
-    // Try to use the preferred orientation, if doesn't fit fallback to the best fit.
-    when {
-        preferredOrientation == Orientation.DOWN && (fitsDown || forceOrientation) ->
-            showPopupWithDownOrientation(anchor, reversed)
-        preferredOrientation == Orientation.UP && (fitsUp || forceOrientation) ->
-            showPopupWithUpOrientation(anchor, containerHeight, reversed)
-        else -> {
-            showPopupWhereBestFits(
-                anchor,
-                fitsUp,
-                fitsDown,
-                reversed,
-                containerHeight,
-            )
+    showPopup(orientation, anchor, recyclerViewHeight, recyclerViewItemCount, containerViewHeight)
+    return true
+}
+
+/**
+ * Show the [PopupWindow] with the required configuration.
+ */
+private fun PopupWindow.showPopup(
+    orientation: Orientation?,
+    anchor: View,
+    recyclerViewHeight: Int,
+    recyclerViewItemCount: Int,
+    containerViewHeight: Int,
+) {
+    val containerHeight =
```

Comment:
Looking at the name of the parameter and the name of the function `getContainerHeight` and the name of passed parameter `containerViewHeight` to be a bit confusing.  `containerViewHeight` is the height of the container if it's drawn as it is, but then the function is calculating the new height. Maybe changing the name from `getContainerHeight` (since we already have it) to `getNewContainerHeight` or `calculateContainerHeight` would be a more clear way to tell what's happening between a passed height and the new calculated height.